### DISCUSSION TITLE
How to read message in a channel from a server that not your own but you're invited.

### DISCUSSION THREAD

**Discussion by anvu03** [discordjs/discord.js#5199] [Q&A] [answered]
Hello,

Is there a way to read messages or listen to new messages from a channel that in a server that not your own but you're a member in that channel? 

Thank you.

**Comment by Androz2091** [comment, marked as answer]
Hello,

Yes, the bot can listen to messages on all channels it has access to, even if you're not the owner of the server (if that is your question). So you only need to add it to the server and use the following code:
```js
const Discord = require('discord.js');
const client = new Discord.Client();

client.on('ready', () => {
     console.log(`Logged in as ${client.user.tag}!`);
});

client.on('message', msg => {
  
    console.log(`New message received from #${msg.channel.name}!`);

});

client.login('token');
```

**Reply by anvu03** [reply]
@Androz2091 , I dont have permission to add a bot to that particular channel, but I'm a member of the channel. Is there still a way that I can read messages from that channel programmatically?

**Reply by Androz2091** [reply]
No, there is no legal way to do this. Only bots are allowed to interact with the Discord API.
